packages/devtools/lib/src/service_extensions.dart

```diff
@@ -10,31 +10,72 @@ import 'ui/analytics_constants.dart' as ga;
 import 'ui/icons.dart';
 
 // Each service extension needs to be added to [_extensionDescriptions].
-class ToggleableServiceExtensionDescription<T> {
-  const ToggleableServiceExtensionDescription._({
-    this.extension,
-    this.description,
+class ToggleableServiceExtensionDescription<T>
+    extends ServiceExtensionDescription {
+  ToggleableServiceExtensionDescription._({
+    Icon icon,
+    @required String extension,
+    @required String description,
+    @required T enabledValue,
+    @required T disabledValue,
+    @required String enabledTooltip,
+    @required String disabledTooltip,
+    @required String gaScreenName,
+    @required String gaItem,
+  }) : super(
+          extension: extension,
+          description: description,
+          icon: icon,
+          values: [enabledValue, disabledValue],
+          tooltips: [enabledTooltip, disabledTooltip],
+          gaScreenName: gaScreenName,
+          gaItem: gaItem,
+        );
+
+  static const enabledValueIndex = 0;
+
+  static const disabledValueIndex = 1;
+
+  T get enabledValue => values[enabledValueIndex];
+
+  T get disabledValue => values[disabledValueIndex];
+
+  String get enabledTooltip => tooltips[enabledValueIndex];
+
+  String get disabledTooltip => tooltips[disabledValueIndex];
+}
+
+class ServiceExtensionDescription<T> {
+  ServiceExtensionDescription({
     this.icon,
-    this.enabledValue,
-    this.disabledValue,
-    this.enabledTooltip,
-    this.disabledTooltip,
+    List<String> displayValues,
+    @required this.extension,
+    @required this.description,
+    @required this.values,
+    @required this.tooltips,
     @required this.gaScreenName,
     @required this.gaItem,
-  });
+  }) : displayValues =
```

Comment:
add
assert(values.length == tooltips.length)

Reply:
This isn't necessarily the case. For example, the platform selector just has one tooltip for the selector ("Override Target Platform"), not one for each of the items in the selection list. For the 2-state extensions, `values.length == tooltips.length` is enforced by the constructor of `ToggleableServiceExtensionDescription`.